### PR TITLE
Introduce a new arguments parser

### PR DESCRIPTION
**Pull request by KingRainbow44**
## Description

Introduces a new parser for application arguments.
Also adds a way to specify the server's language. (so no user prompts are required)

## Issues fixed by this PR

Fixes the major issue of #686.

## Type of changes

- [x] Bug fix
- [x] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: src/main/java/emu/grasscutter/utils/Utils.java [hunk 1]
```diff
@@ -409,4 +409,4 @@ public static <T> T drawRandomListElement(List<T> list, List<Integer> probabilit
     public static <T> T drawRandomListElement(List<T> list) {
         return drawRandomListElement(list, null);
     }
-}
+}
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/utils/Utils.java]: :/

File: src/main/java/emu/grasscutter/utils/StartupArguments.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        "-gachamap", StartupArguments::generateGachaMap,
+        "-version", StartupArguments::printVersion,
+        "-debug", StartupArguments::enableDebug,
+        "-lang", parameter -> {
+            Grasscutter.setPreferredLanguage(parameter); return false;
+        },
+
+        // Aliases.
+        "-v", StartupArguments::printVersion,
+        "-debugall", StartupArguments::enableDebug
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/utils/StartupArguments.java]: `-debugall` is not an alias for `-debug`, `-debug` currently sets the level to `Level.INFO` and `-debugall` to `Level.DEBUG`

**Review by Birdulon — CHANGES_REQUESTED** [review]

File: src/main/java/emu/grasscutter/utils/StartupArguments.java (hunk 2 above)
Comment (KingRainbow44) [reply src/main/java/emu/grasscutter/utils/StartupArguments.java]: the first *five* times I look at it, I swore it was the same

File: src/main/java/emu/grasscutter/utils/Utils.java (hunk 1 above)
Comment (KingRainbow44) [reply src/main/java/emu/grasscutter/utils/Utils.java]: not me, it was IntelliJ

**Review by Birdulon — APPROVED** [review]

**Review by 323083273 — COMMENTED** [review]
src/main/java/emu/grasscutter/Grasscutter.java

**Pull request merged**